### ISSUE TITLE
Building of example fails

### ISSUE THREAD

**Issue by anates** [slowriot/libtelegram#4]
Hei,
when trying to compile the echo-example, I get
```
Compiling examples/echo.cpp...                                              
In file included from /usr/include/boost/asio/write.hpp:927:0,
                 from /usr/include/boost/asio/buffered_write_stream.hpp:29,
                 from /usr/include/boost/asio/buffered_stream.hpp:22,
                 from /usr/include/boost/asio.hpp:41,
                 from include/boost/cgi/fcgi/acceptor_service_impl.hpp:16,
                 from include/boost/cgi/fcgi/request_acceptor_service.hpp:21,
                 from include/boost/cgi/fcgi/acceptor.hpp:14,
                 from include/boost/cgi/fcgi.hpp:15,
                 from include/libtelegram/listeners/fcgi.h:6,
                 from include/libtelegram/listener.h:5,
                 from include/libtelegram/libtelegram.h:4,
                 from examples/echo.cpp:3:
/usr/include/boost/asio/impl/write.hpp: In instantiation of ‘std::size_t boost::asio::detail::write_buffer_sequence(SyncWriteStream&, const ConstBufferSequence&, const ConstBufferIterator&, CompletionCondition, boost::system::error_code&) [with SyncWriteStream = boost::cgi::common::basic_connection<boost::cgi::common::tags::shareable_tcp_socket>; ConstBufferSequence = std::vector<boost::asio::const_buffer>; ConstBufferIterator = __gnu_cxx::__normal_iterator<const boost::asio::const_buffer*, std::vector<boost::asio::const_buffer> >; CompletionCondition = boost::asio::detail::transfer_all_t; std::size_t = long unsigned int]’:
/usr/include/boost/asio/impl/write.hpp:69:39:   required from ‘std::size_t boost::asio::write(SyncWriteStream&, const ConstBufferSequence&, CompletionCondition, boost::system::error_code&, typename std::enable_if<boost::asio::is_const_buffer_sequence<ConstBufferSequence>::value>::type*) [with SyncWriteStream = boost::cgi::common::basic_connection<boost::cgi::common::tags::shareable_tcp_socket>; ConstBufferSequence = std::vector<boost::asio::const_buffer>; CompletionCondition = boost::asio::detail::transfer_all_t; std::size_t = long unsigned int; typename std::enable_if<boost::asio::is_const_buffer_sequence<ConstBufferSequence>::value>::type = void]’
include/boost/cgi/fcgi/client.hpp:90:69:   required from here
/usr/include/boost/asio/impl/write.hpp:53:9: error: no matching function for call to ‘boost::cgi::common::basic_connection<boost::cgi::common::tags::shareable_tcp_socket>::write_some(boost::asio::detail::consuming_buffers<boost::asio::const_buffer, std::vector<boost::asio::const_buffer>, __gnu_cxx::__normal_iterator<const boost::asio::const_buffer*, std::vector<boost::asio::const_buffer> > >::prepared_buffers_type, boost::system::error_code&)’
         tmp.consume(s.write_some(tmp.prepare(max_size), ec));
```
while running GCC 7.3.1 and boost 1.66.0. What can I do to fix that?
Thanks!

**Comment by slowriot** [comment]
Thanks for the report. Looks like an issue in boost cgi (which isn't formally part of boost) not being fully compatible with the latest version. I'll investigate.

Meanwhile, if you're only using the polling listener, you can omit boost cgi entirely by disabling the fastcgi listener - define TELEGRAM_NO_LISTENER_FCGI project-wide or before you first include listener.h.

**Comment by anates** [comment]
If possible, please check for both 1.66 and 1.67, thanks!

**Comment by slowriot** [comment]
Should be resolved by ba7b58a3827399c8642a413d09dd39bd2b174900
